### PR TITLE
[v2] Edit packages peerDependencies to match Gatsby v2-alpha

### PR DESCRIPTION
**Pull request by ryanditjia**

### CODE REVIEW

File: packages/gatsby-link/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
   "license": "MIT",
   "main": "index.js",
   "peerDependencies": {
-    "gatsby": "^1.0.0"
+    "gatsby": "^1.0.0 || ^2.0.0-alpha"
```
Comment (pieh) [inline packages/gatsby-link/package.json]: Is v2 version of `gatsby-link ` actually working with v1? There were some changes (started here - https://github.com/gatsbyjs/gatsby/pull/4714 ) that make me not sure if we should keep v1 here

Comment (nihgwu) [reply packages/gatsby-link/package.json]: I also have the same doubts, I think the semver for `gatsby-link` is wrong, as I encountered with the prefixPath broken again and again, and I don't know which part is wrong, so I choose to use `resolutions` to lock the version to the specific one `gatsby` depends on

You can see that the version for `next` is behind `latest`, that's strange, the same for `gatsby-cli`

**Comment by ryanditjia** [comment]
Should we tag gatsby-link post #4714 as 2.0.0?

**Comment by nihgwu** [comment]
maybe we can resolve #5732 before bumping to 2.0?

**Pull request closed without merging**